.github/workflows/test.yaml

```diff
@@ -21,28 +10,22 @@ jobs:
       matrix:
         os: ["ubuntu-latest", "windows-latest", "macos-latest"]
         environment: [test310, test311, test312, test313]
+    
     steps:
-      - uses: actions/checkout@v4
-      - uses: prefix-dev/setup-pixi@v0.8.8
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683  # v4.2.2
+      - name: Setup pixi
+        uses: prefix-dev/setup-pixi@19eac09b398e3d0c747adc7921926a6d802df4da  # v0.8.8
         with:
-          pixi-version: latest
           cache: true
-          # auth-host: prefix.dev
-          # auth-token: ${{ secrets.PREFIX_DEV_TOKEN }}
-      - run: pixi run --environment ${{ matrix.environment }} test
-  lint:
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-    steps:
-      - uses: actions/checkout@v4
-      - uses: prefix-dev/setup-pixi@v0.8.8
-        with:
-          pixi-version: latest
-          cache: true
-          # auth-host: prefix.dev
-          # auth-token: ${{ secrets.PREFIX_DEV_TOKEN }}
-      - run: pixi run --environment test310 lint
+
+      - name: Tests
+        run: >
+          pixi run --environment ${{ matrix.environment }} test
+        
+      - name: Lint
+        if: contains(${{ matrix.os }}, "ubuntu") && ${{ matrix.environment }} == "test310"
```

Comment:
This is not working yet. I need to check GHA docs. However, if we go the route proposed in #66, we should remove linting from here anyway.

Reply:
Why not keep it the way it was? it worked, and I prefer the lint to be a separate job anyway.

Putting it in the matrix, and then selecting out only part the matrix seems like complication for no reason.

But if it's made a pre-commit hook, then as you say, a moot point.